### PR TITLE
Feature/666 ce iterator support

### PR DESCRIPTION
**Pull request by rallen2010**
- Community edition support for BPMN Iterators and Loops
- Also Adds JSON typed Process Variables to CE

### CODE REVIEW

File: src/views/flow_instance_details_vw.sql
Comment (commi235) [inline src/views/flow_instance_details_vw.sql:1]: Probably keep this for Community Edition

File: src/views/flow_instance_details_vw2.sql
Comment (commi235) [inline src/views/flow_instance_details_vw2.sql:1]: should be removed from all editions

File: src/views/flow_instance_details_vw3.sql
Comment (commi235) [inline src/views/flow_instance_details_vw3.sql:1]: This seems to be a Enterprise Edition only view.

File: src/views/flow_subflows_vw.sql
Comment (commi235) [inline src/views/flow_subflows_vw.sql:1]: Check for stuff to be removed

File: src/views/flow_viewer_details_vw.sql
Comment (commi235) [inline src/views/flow_viewer_details_vw.sql:1]: Check if this file is needed.

**Review by commi235 — CHANGES_REQUESTED** [review]
see comments

File: src/views/flow_instance_details_vw2.sql
Comment (rallen2010) [reply src/views/flow_instance_details_vw2.sql:1]: removed

File: src/views/flow_instance_details_vw.sql
Comment (rallen2010) [reply src/views/flow_instance_details_vw.sql:1]: Kept, with added note to maintainers to also update the EE version if structure changed.

File: src/views/flow_instance_details_vw3.sql
Comment (rallen2010) [reply src/views/flow_instance_details_vw3.sql:1]: Moved to EE

File: src/views/flow_subflows_vw.sql
Comment (rallen2010) [reply src/views/flow_subflows_vw.sql:1]: Not sure I understand the problem here....

File: src/views/flow_viewer_details_vw.sql [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    join flow_diagrams dgrm
+      on prdg.prdg_dgrm_id = dgrm.dgrm_id
+    join iteration_paths ip
+      on ip.prcs_id = prcs.prcs_id
+    left join flow_process_variables prov
+      on prov.prov_prcs_id  = prcs.prcs_id
+     and prov.prov_var_name = 'BUSINESS_REF'
+     and prov.prov_var_type = 'VARCHAR2'
+     and prov.prov_scope    = 0
+     where prcs.prcs_id = 8416;
```
Comment (commi235) [inline src/views/flow_viewer_details_vw.sql]: process id included in the view.

File: src/views/flow_viewer_details_vw.sql [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+  select sbfl_prcs_id as prcs_id
+       , sbfl_dgrm_id as dgrm_id
+       , sbfl_diagram_level as diagram_level
+       , sbfl_iteration_path as iter_path
+       , listagg(sbfl_current, ':') within group (order by sbfl_current) as bpmn_ids
+    from flow_subflows
+   where sbfl_current is not null
+     and sbfl_status = 'error'
+group by sbfl_prcs_id, sbfl_dgrm_id, sbfl_diagram_level, sbfl_iteration_path
+), iteration_paths as (
```
Comment (commi235) [inline src/views/flow_viewer_details_vw.sql]: should we slim this down for community edition?

**Review by commi235 — COMMENTED** [review]
please check new comments

File: src/views/flow_viewer_details_vw.sql (hunk 1 above)
Comment (rallen2010) [reply src/views/flow_viewer_details_vw.sql]: This file should be deleted - we don't use it.  (was an idea we tested and decided not to use - the viewer is still driven from flow_instance_details_vw, as before.   It now has 2 versions, one simple one in CE and a more complex one in EE - d we fixed the file locations / names yesterday...

File: src/views/flow_viewer_details_vw.sql (hunk 2 above)
Comment (rallen2010) [reply src/views/flow_viewer_details_vw.sql]: file / view should be deleted

**Review by commi235 — APPROVED** [review]

**Pull request merged**